### ISSUE TITLE
Sending message to a Handler on a dead thread

### ISSUE THREAD

**Issue by SparklingLukas** [googleads/googleads-mobile-unity#733]
Using Unity 2017.4.3f1 and AdMob 3.15.0 I get the follow error in Logcat. 
This error seems be caused by calling `InterstitialAd.Destroy()` inside the `InterstitialAd.OnAdClosed` eventhandler.

I can reproduce it on an Acer A1-840FHD (API 19) but not on Wiko View2 (API 26).

Edit: After some more testing it seems to happen on the second, third etc Interstitial I close but not on the first.

> W/MessageQueue: Handler (android.os.Handler) {221e7290} sending message to a Handler on a dead thread
>     java.lang.RuntimeException: Handler (android.os.Handler) {221e7290} sending message to a Handler on a dead thread
>         at android.os.MessageQueue.enqueueMessage(MessageQueue.java:336)
>         at android.os.Handler.enqueueMessage(Handler.java:626)
>         at android.os.Handler.sendMessageAtTime(Handler.java:595)
>         at android.os.Handler.sendMessageDelayed(Handler.java:566)
>         at android.os.Handler.sendEmptyMessageDelayed(Handler.java:530)
>         at android.os.Handler.sendEmptyMessage(Handler.java:515)
>         at com.google.android.gms.ads.internal.video.exoplayer1.f.b(:com.google.android.gms.dynamite_adsdynamite@12874005@12.8.74 (000700-204998136):181)
>         at com.google.android.gms.ads.internal.webview.t.v(:com.google.android.gms.dynamite_adsdynamite@12874005@12.8.74 (000700-204998136):22)
>         at com.google.android.gms.ads.internal.webview.j.onPageFinished(:com.google.android.gms.dynamite_adsdynamite@12874005@12.8.74 (000700-204998136):84)
>         at com.android.webview.chromium.WebViewContentsClientAdapter.onPageFinished(WebViewContentsClientAdapter.java:454)
>         at com.android.org.chromium.android_webview.AwContentsClient$AwWebContentsObserver$1.run(AwContentsClient.java:73)
>         at android.os.Handler.handleCallback(Handler.java:733)
>         at android.os.Handler.dispatchMessage(Handler.java:95)
>         at android.os.Looper.loop(Looper.java:149)
>         at android.app.ActivityThread.main(ActivityThread.java:5061)
>         at java.lang.reflect.Method.invokeNative(Native Method)
>         at java.lang.reflect.Method.invoke(Method.java:515)
>         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:794)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:610)
>         at dalvik.system.NativeStart.main(Native Method)

**Comment by rampara** [comment]
@SparklingLukas I'm having trouble reproducing the issue. Are you able to reproduce the issue on any device other than a Acer A1-840FHD.

**Comment by SparklingLukas** [comment]
@rampara I have tested again and I can also reproduce it on Samsung Note 3 (API 23) and Samsung S6 (API 24). 

My steps to reproduce is: 

1. Start game
2. Wait for interstitial to be available, so play the game a bit
3. Show interstitial
4. When interstitial is done, click the round close button at top right
5. Interstitial closes
6. Error shows up in logcat

Below is the stacktrace from the Note 3. I included a little bit more stacktrace than previous message.

> 07-30 16:57:58.710 15150-16451/com.sparklingsocietysims.townville E/chromium: [ERROR:gl_context_virtual.cc(39)] Trying to make virtual context current without decoder.
> 07-30 16:55:13.620 15150-15150/com.sparklingsocietysims.townville I/Unity: windowFocusChanged: true
> 07-30 16:55:13.630 15150-15150/com.sparklingsocietysims.townville W/Ads: #004 The webview is destroyed. Ignoring action.
>     #004 The webview is destroyed. Ignoring action.
> 07-30 16:55:13.640 15150-15150/com.sparklingsocietysims.townville I/Unity: onResume
> 07-30 16:55:13.680 15150-15150/com.sparklingsocietysims.townville I/Ads: Starting ad request.
> 07-30 16:55:13.730 15150-15150/com.sparklingsocietysims.townville W/Ads: #004 The webview is destroyed. Ignoring action.
>     #004 The webview is destroyed. Ignoring action.
> 07-30 16:55:13.830 15150-15150/com.sparklingsocietysims.townville W/MessageQueue: Handler (android.os.Handler) {1e0fd3c} sending message to a Handler on a dead thread
>     java.lang.IllegalStateException: Handler (android.os.Handler) {1e0fd3c} sending message to a Handler on a dead thread
>         at android.os.MessageQueue.enqueueMessage(MessageQueue.java:543)
>         at android.os.Handler.enqueueMessage(Handler.java:631)
>         at android.os.Handler.sendMessageAtTime(Handler.java:600)
>         at android.os.Handler.sendMessageDelayed(Handler.java:570)
>         at android.os.Handler.sendEmptyMessageDelayed(Handler.java:534)
>         at android.os.Handler.sendEmptyMessage(Handler.java:519)
>         at com.google.android.gms.ads.internal.video.exoplayer1.f.b(:com.google.android.gms.dynamite_adsdynamite@12874019@12.8.74 (040300-204998136):177)
>         at com.google.android.gms.ads.internal.webview.t.v(:com.google.android.gms.dynamite_adsdynamite@12874019@12.8.74 (040300-204998136):22)
>         at com.google.android.gms.ads.internal.webview.j.onPageFinished(:com.google.android.gms.dynamite_adsdynamite@12874019@12.8.74 (040300-204998136):84)
>         at com.android.webview.chromium.WebViewContentsClientAdapter.onPageFinished(WebViewContentsClientAdapter.java:224)
>         at org.chromium.android_webview.AwContentsClientCallbackHelper$MyHandler.handleMessage(AwContentsClientCallbackHelper.java:72)
>         at android.os.Handler.dispatchMessage(Handler.java:102)
>         at android.os.Looper.loop(Looper.java:158)
>         at android.app.ActivityThread.main(ActivityThread.java:7225)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
> 07-30 16:55:13.830 15150-15150/com.sparklingsocietysims.townville W/ExoPlayerImplInternal: Sent message(1) after release. Message ignored.
> 07-30 16:55:13.850 15150-15150/com.sparklingsocietysims.townville I/Timeline: Timeline: Activity_idle id: android.os.BinderProxy@f815067 time:1655015
> 07-30 16:55:14.680 15150-15150/com.sparklingsocietysims.townville I/ExoPlayerImpl: Init 1.3.1

**Comment by SparklingLukas** [comment]
@rampara  I think I found the reason why you couldn't repro it. 

I created a repro project with only our AdMob provider and at first I didn't get the above error. But I noticed that I got different looking interstitials, comparing to the live game. In the repro I got the simple/cheap looking ads; no video's but just a static image with some brand awareness and a close (X) button.

Then I changed the project settings with the game's bundle name and sign it with the game's keystore. The game is quite popular with 10M+ downloads.
Now I got the fancy interstitials; a fancy video about some game with ratings and # of downloads, with install button and close button.

This is the catch: the simple interstitial **did not** throw the error, while the fancy game video **did**.

@rampara I could provide the repro Unity project but I do not want to share it in a public space.
